scripts/patch.sh

```diff
@@ -0,0 +1,17 @@
+stitch() {
```

Comment:
should shebang/errexit/pipefail. 

I'm also unsure why this should be in shell and not in golang.

Reply:
> shebang/errexit/pipefail

im a shell noob - what does this mean? 

> I'm also unsure why this should be in shell and not in golang.

I think the only other option is to run it after the protoc-gen-go-fastreflection plugin runs, but wasn't sure if we wanted to have it as part of that code. Since we are running protoc-gen-go (google's generator) from shell, i can't hook into their code as id like

Reply:
I believe that this will run in whatever `/bin/sh` is on the system, which is `bash` on most systems, but is `dash` on ubuntu/debian? and versioning/mode that `/bin/sh` is can be a bit rough. If you end up running this on recent versions of macOS, I think it ends up being `zsh` (but I'm not sure at all), which is just confusing. Setting a shebang line (the first line of the file being `#!/bin/bash` ensures that the file is run with `bash` even on systems where this isn't the default. 

All shells (I think?) also have a curious semantic where they only return an error if the last command in the file is an error, but will not error if intermediate commands have errors putting the statement `set +o errexit` changes this (I believe `set -e` is the same as this). 

Even with errexit, if you have a `|` expression in a shell script, the command doesn't count as an error unless the final command in the pipeline fails. So this can lead to errors getting hidden. In bash `set +o pipefile` changes this behavior, but it's typically not avalible in `dash` and canonical `sh` implementations. 

Anyway, shell is terrible, and I'm always tough of shell scripts because of this and I try to avoid writing them when possible: 
- http://redsymbol.net/articles/unofficial-bash-strict-mode/
- https://olivergondza.github.io/2019/10/01/bash-strict-mode.html

Reply:
If we can avoid shell altogether here that sounds preferable. Can this be run as part of the call to protoc somehow? Or will all apps need to have a script like this?

Reply:
should be fine to run it during the call to our protoc plugin - but i think that requires not to use docker. doesnt look like i have access to host file system when our plugin is called, only file system in docker 🤷🏻 maybe there is a way around this but im not sure